### PR TITLE
Document MathHelper.binarySearch

### PR DESCRIPTION
**Pull request by enbrain**

### CODE REVIEW

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 1]
```diff
@@ -86,9 +86,29 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
...
 		ARG 0 value
 	METHOD method_15360 binarySearch (IILjava/util/function/IntPredicate;)I
-		ARG 0 start
-		ARG 1 end
-		ARG 2 leftPredicate
+		COMMENT Finds the minimum value in {@code [min, max)} that satisfies the
+		COMMENT monotonic {@code predicate}.
+		COMMENT
+		COMMENT <p>The {@code predicate} must be monotonic i.e. for any {@code a} and
+		COMMENT {@code b} in {@code [min, max)} such that {@code a <= b}, if
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping]:
```suggestion
		COMMENT {@code predicate.test(a)} is {@code true},
```

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 2]
```diff
@@ -86,9 +86,29 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
 	METHOD method_15357 floor (D)I
 		ARG 0 value
 	METHOD method_15360 binarySearch (IILjava/util/function/IntPredicate;)I
-		ARG 0 start
-		ARG 1 end
-		ARG 2 leftPredicate
+		COMMENT Finds the minimum value in {@code [min, max)} that satisfies the
+		COMMENT monotonic {@code predicate}.
+		COMMENT
+		COMMENT <p>The {@code predicate} must be monotonic i.e. for any {@code a} and
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping]:
```suggestion
		COMMENT <p>The {@code predicate} must be monotonic, i.e. if for any {@code a},
```

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 3]
```diff
@@ -86,9 +86,29 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
...
+		COMMENT {@code b} in {@code [min, max)} such that {@code a <= b}, if
+		COMMENT {@code predicate(a)} is {@code true}, then {@code predicate(b)} must
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping]:
```suggestion
		COMMENT then for all {@code b > a}, {@code predicate.test(b)} must
```

**Review by liach — APPROVED** [review]
if there's specified behavior if the function is not monotonic (like throwing iae), we can document it too

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**